src/LinuxTracing/StackSlice.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#ifndef LINUX_TRACING_STACK_SLICE_H_
+#define LINUX_TRACING_STACK_SLICE_H_
+
+#include <absl/types/span.h>
+
+#include <memory>
+
+namespace orbit_linux_tracing {
+
+// This class holds a copy of some stack data from collected from the target process.
+// The class must have a single owner, to avoid unnecessary copies. Instead of passing pointers or
+// references of this class around, `StackSliceView` should be used instead.
+class StackSlice {
+ public:
+  StackSlice(uint64_t start_address, uint64_t size, std::unique_ptr<char[]>&& data)
+      : start_address_{start_address}, size_{size}, data_{std::move(data)} {}
+
+  [[nodiscard]] uint64_t GetStartAddress() const { return start_address_; }
+  [[nodiscard]] uint64_t GetSize() const { return size_; }
+  [[nodiscard]] const char* GetData() const { return data_.get(); }
+
+ private:
+  uint64_t start_address_;
+  uint64_t size_;
+  std::unique_ptr<char[]> data_;
+};
+
+// This class is a "view" of a `StackSlice`, which contains a bare pointer (as "absl::Span") to the
+// data field of the `StackSlice` this corresponds to. The lifetime of this class is bound to the
+// lifetime of the `StackSlice` it is pointing to.
+class StackSliceView {
```

Comment:
And isn't this added for the purpose of `LibunwindstackUnwinder`'s interface? In that case I think it should be in `LibunwindstackUnwinder.h` like `LibunwindstackResult`?

Reply:
Isn't that kind of against the one file per class policy?

Reply:
They are not unrelated classes or "peers". The purpose of those two classes is only to be part of `LibunwindstackUnwinder`'s interface, so in my opinion it's clearer and more convenient if they are in the same file as the interface they support.